pkg/component/observability/opentelemetry/collector/constants/constants.go

```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package constants
+
+const (
+	// OpenTelemetryCollectorResourceName is the name of the OpenTelemetry Collector resource.
+	OpenTelemetryCollectorResourceName = "opentelemetry-collector"
+	// DeploymentName is the name that the OtelOperator will for the Collector deployment.
```

Comment:
Nit: same here for `otel-oeprator` vs `OtelOperator`.
```suggestion
	// DeploymentName is the name that the otel-operator will use for the OpenTelemetry Collector deployment.
```